### PR TITLE
Migrate to Zeebe 1.0.0 API

### LINKED ISSUE #2169: Migrate to new Zeebe Endpoint APIs (for deploy and run)
__Is your feature request related to a problem? Please describe.__

The Camunda Modeler (Zeebe Tabs) offers the option to "deploy" or "run" (start instance) of a Zeebe process using the "Deploy and run" modal menues. 

With the 1.0.0 the Zeebe Broker APIs will change (in particular: `workflow` will be renamed to `process`). Hence the "Deploy and run" feature will not work anymore. (This can be tested using the [`1.0.0-alpha3` release of Zeebe Broker or later](https://github.com/camunda-cloud/zeebe/releases)).

__Describe the solution you'd like__

* Update to a later version of https://github.com/camunda-community-hub/zeebe-client-node-js which supports the new `1.0.0` broker API endpoints. 
* Release this change on 11.05.2021 (together with Zeebe 1.0.0 stable).
* Document that Camunda-Modeler v4.8.0 (tbd) supports only Zeebe 1.0.0+ for deploy and run features

__Describe alternatives you've considered__

n/a

__Additional context__

cc. @menski 
cc. @jwulf. @jwulf , as far as I understood the to be used zeebe-node-js version would be based on the JS (not the native gRPC) implementation right? Because otherwise, we would run into problems (because of Electron incompatability, we so far were not able to use the native gRPC implementation)

### PR DESCRIPTION
**Pull request by barmac**
Closes #2169

### CODE REVIEW

**Comment by barmac** [comment]
This prints the grpc error for unsupported Zeebe version:

```
Method not found: gateway_protocol.Gateway/DeployProcess [ deploy-error ]
```

We could overwrite it to something less cryptic, but we haven't done it with the deployment errors yet.

File: app/lib/zeebe-api.js [hunk 1]
```diff
@@ -143,7 +143,7 @@ class ZeebeAPI {
 
       return {
         success: false,
-        response: err
+        response: asSerializedError(err)
```
Comment (nikku) [inline app/lib/zeebe-api.js:146]: Good catch on this one :clap:

**Comment by pinussilvestrus** [comment]
> We could overwrite it to something less cryptic, but we haven't done it with the deployment errors yet.

can we ensure this error message we receive from the broker won't change when we override it? I'd be fine with just keeping displaying the raw error.

**Comment by pinussilvestrus** [comment]
Using an old Zeebe version (I use v0.26), shouldn't it display it as an error in the Modal? Instead, it allows me to execute the deploy, but then show the error in the Notification. 

Kapture 2021-04-27 at 17 30 37

**Comment by nikku** [comment]
@pinussilvestrus The fact that it fails on deploy only is intended for the moment.

Obviously you could argue that we use the topology features (https://github.com/camunda/camunda-modeler/pull/2228, separate PR for zeebe will follow) to detect the miss-match early. My proposal would be that we do not block this PR because of it, though.

At the moment the error is being displayed and it is displayed clear enough for people to post it somewhere and get help.

Depending on @MaxTru work in version land we could decide to follow up on this PR.

**Comment by nikku** [comment]
Let us not override / fine tune the deployment errors for now. Whatever we do it will shadow the underlying issue (https://github.com/camunda/camunda-modeler/pull/2231#issuecomment-827699222).

**Comment by pinussilvestrus** [comment]
Agree with all 👍

**Comment by barmac** [comment]
> Using an old Zeebe version (I use `v0.26` via Camunda Cloud), shouldn't it display it as an error in the Modal? Instead, it allows me to execute the deploy (without error), but then show the error in the Notification.
> 
> Kapture 2021-04-27 at 17 30 37

Once Max' pr with version detection is done, we could switch the method for connection check. Still, it's not necessary to support Zeebe 1.

Can we merge this increment and follow up later?

**Review by pinussilvestrus — APPROVED** [review]
👍

**Pull request merged**